Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -96,14 +156,45 @@ Serializer::TypedValue Serializer::DecodeTypedValue(const uint8_t* bytes,
     abort();
   }
 
+  Serializer::TypedValue retval{FieldValue::Type::Null,
+                                google_firestore_v1beta1_Value_init_default};
   switch (tag) {
     case google_firestore_v1beta1_Value_null_value_tag:
-      return Serializer::TypedValue{
-          FieldValue::Type::Null, google_firestore_v1beta1_Value_init_default};
+      retval.type = FieldValue::Type::Null;
+      DecodeNull(&stream);
+      break;
+    case google_firestore_v1beta1_Value_boolean_value_tag:
+      retval.type = FieldValue::Type::Boolean;
+      retval.value.boolean_value = DecodeBool(&stream);
+      break;
+
     default:
       // TODO(rsgowman): figure out error handling
       abort();
   }
+
+  return retval;
+}
+
+bool operator==(const Serializer::TypedValue& lhs,
+                const Serializer::TypedValue& rhs) {
+  if (lhs.type != rhs.type) {
+    return false;
+  }
+
+  switch (lhs.type) {
+    case firebase::firestore::model::FieldValue::Type::Null:
+      FIREBASE_DEV_ASSERT(lhs.value.null_value ==
+                          google_protobuf_NullValue_NULL_VALUE);
+      FIREBASE_DEV_ASSERT(rhs.value.null_value ==
+                          google_protobuf_NullValue_NULL_VALUE);
+      return true;
+    case firebase::firestore::model::FieldValue::Type::Boolean:
```

Comment:
You already use `FieldValue::Type::Boolean` in code above. Maybe `using` those at the top in this .cc file?

Reply:
Turns out, I'm already `using` FieldValue, so removing at least the firebase:firestore:model is certainly appropriate. (Oh I see what I did; this was the code I moved from the header file.) Done.